### ISSUE TITLE
How to allow empty value for `group-by`?

### ISSUE THREAD

**Issue by fdw** [nils-werner/crestic#30]
Hi! 🙂

I'd like to set `--group-by = ''`, as described in the [documentation](https://restic.readthedocs.io/en/stable/060_forget.html#removing-snapshots-according-to-a-policy) to reset the original grouping. However, I haven't found a way that works with `crestic` - if I don't set a value, like `group-by: `, the config can't be parsed correctly. If I use `group-by: ''` explicitly, `restic` is confronted with `--group-by ''''`.

Any ideas or help?

**Comment by nils-werner** [comment]
Hm, that's a tough one... I don't have a solution I'm afraid

**Comment by fdw** [comment]
Would it be possible to use `''` and not add quotes if it's an empty string? I think that would be the easiest solution.

**Comment by nils-werner** [comment]
No, because an empty string is used as an option flag (`--foo` without any following value)

**Comment by mariuszste** [comment]
only sensible way to do this would be to know what flags take values and what flags don't. But that will add a lot of extra complexity and maintenance needed.

Or maybe we could support different config formats like for exmaple json, that would allow us to do something like
```json
{"foo": true}

**Comment by nils-werner** [comment]
Yeah I don't like either. Knowing about all options is a lot of maintenance, and more formats will needlessly increase complexity.

Maybe a `raw` key that lets you set any list parameters that you want?

```
raw:
    --group-by
    ""
```

**Comment by nils-werner** [comment]
Oh and [ConfigParser allows to be set up](https://docs.python.org/3/library/configparser.html#supported-ini-file-structure)  to have

```
[section]
key_without_value
empty_string_value =
```

If we enable this behavior, maybe we don't have to abuse empty string values for options without values.

**Comment by nils-werner** [comment]
Oh and [ConfigParser allows to be set up](https://docs.python.org/3/library/configparser.html#supported-ini-file-structure)  to allow

```
[section]
key_without_value
empty_string_value =
```

If we enable this behavior, maybe we don't have to abuse empty string values for options without values.

**Comment by nils-werner** [comment]
OK, I've implemented that behaviour in `master`, can you check if this works in your setups? The rules are now

```
[section]
multi:
   is
   repeated
empty:
switch
```

leads to

```
--multi is --multi repeated --empty "" --switch
```

**Comment by dumbasPL** [comment]
haven't tested, but I'm pretty sure this would be a breaking change for existing setups since `empty: ` used to create a flag but now it adds extra `""`.

Having a backup scrip/software break on you when you don't expect it is an absolute nightmare. Found out the hard way in the past...

**Comment by nils-werner** [comment]
Yeah don't worry, I have that on my radar :-)

**Comment by nils-werner** [comment]
Actually, I'm inclined to just bump the major version number and create a changelog with a warning. I don't want to go through an entire deprecation cycle for such a tiny tool as this...

**Comment by mariuszste** [comment]
just a note: in major version 0 aka 0.x.x breaking changes are allowed on minor version changes. at least that's what most people seem to agree on. bumping it to 1.0.0 would signify that It's "production ready"

**Comment by nils-werner** [comment]
You mean [this paragraph](https://semver.org/#spec-item-4)? I don't read that as "production ready", but I see what you mean...

I think I'll still bump the major version, because

 1. this right now is the public API that I had in mind, and I don't plan on changing it
 2. this change can really mess with (a small group of) people, so I want to make sure there is no misunderstanding

**Comment by mariuszste** [comment]
> You mean [this paragraph](https://semver.org/#spec-item-4)? I don't read that as "production ready", but I see what you mean...

I mostly mean [point 4](https://semver.org/#spec-item-4) (anything might break/change at any moment in 0.x.y) and [point 5](https://semver.org/#spec-item-4) (1.0.0, from this point onward, any breaking change requires a major version bump)

If you believe the api is/will be stable then going for a 1.0.0 release is indeed a good idea.

**Comment by nils-werner** [comment]
I've just released 0.7.0 which introduces the [new syntax](https://nils-werner.github.io/crestic/config/options.html#option-switches) and emits a deprecationwarning for the old syntax. In a month or so I'll release 1.0.0 which will change the behaviour of the old syntax.
